crates/ruff_python_formatter/src/comments/placement.rs

```diff
@@ -18,85 +18,88 @@ pub(super) fn place_comment<'a>(
     comment: DecoratedComment<'a>,
     locator: &Locator,
 ) -> CommentPlacement<'a> {
-    // Handle comments before and after bodies such as the different branches of an if statement.
-    let comment = if comment.line_position().is_own_line() {
-        handle_own_line_comment_around_body(comment, locator)
-    } else {
-        handle_end_of_line_comment_around_body(comment, locator)
-    };
-
-    // Change comment placement depending on the node type. These can be seen as node-specific
-    // fixups.
-    comment.or_else(|comment| match comment.enclosing_node() {
-        AnyNodeRef::Parameters(arguments) => {
-            handle_parameters_separator_comment(comment, arguments, locator)
-                .or_else(|comment| handle_bracketed_end_of_line_comment(comment, locator))
-        }
-        AnyNodeRef::Arguments(_) | AnyNodeRef::TypeParams(_) => {
-            handle_bracketed_end_of_line_comment(comment, locator)
-        }
-        AnyNodeRef::Comprehension(comprehension) => {
-            handle_comprehension_comment(comment, comprehension, locator)
-        }
-        AnyNodeRef::ExprAttribute(attribute) => handle_attribute_comment(comment, attribute),
-        AnyNodeRef::ExprBinOp(binary_expression) => {
-            handle_trailing_binary_expression_left_or_operator_comment(
-                comment,
-                binary_expression,
-                locator,
-            )
-        }
-        AnyNodeRef::Keyword(_) => handle_dict_unpacking_comment(comment, locator),
-        AnyNodeRef::ExprDict(_) => handle_dict_unpacking_comment(comment, locator)
-            .or_else(|comment| handle_bracketed_end_of_line_comment(comment, locator)),
-        AnyNodeRef::ExprIfExp(expr_if) => handle_expr_if_comment(comment, expr_if, locator),
-        AnyNodeRef::ExprSlice(expr_slice) => handle_slice_comments(comment, expr_slice, locator),
-        AnyNodeRef::ExprStarred(starred) => {
-            handle_trailing_expression_starred_star_end_of_line_comment(comment, starred)
-        }
-        AnyNodeRef::ExprSubscript(expr_subscript) => {
-            if let Expr::Slice(expr_slice) = expr_subscript.slice.as_ref() {
+    handle_parenthesized_comment(comment, locator)
+        .or_else(|comment| handle_end_of_line_comment_around_body(comment, locator))
+        .or_else(|comment| handle_own_line_comment_around_body(comment, locator))
+        .or_else(|comment| match comment.enclosing_node() {
+            AnyNodeRef::Parameters(arguments) => {
+                handle_parameters_separator_comment(comment, arguments, locator)
+                    .or_else(|comment| handle_bracketed_end_of_line_comment(comment, locator))
+            }
+            AnyNodeRef::Arguments(_) | AnyNodeRef::TypeParams(_) => {
+                handle_bracketed_end_of_line_comment(comment, locator)
```

Comment:
Hmm, but removing this leads to a bunch of fixture changes!

Reply:
Can we look into why it is still necessary? I would expect that we can remove all the open parentheses handling code. Or is it because these need to be attached as dangling rather than leading comments?

Reply:
Yeah, these need to be attached as dangling. We want to format like:

```python
f(  # comment
    a,
)
```

Instead of:

```python
f(
    # comment
    a,
)
```

Though in actuality, as of this PR, if we attach it as leading to `a`, it would now be rendered as a parenthesized comment on `a`, rather than on the arguments:

```python
f(
    (  # comment
        a
    ),
)
```